false positive on E1101 with contextmanager decorated generators

I have this little snippet:

```
import contextlib

@contextlib.contextmanager
def context_manager():
    try:
        yield
    finally:
        pass

cm = context_manager()
cm.__enter__()
cm.__exit__(None, None, None)
```

pylint signals the "direct" access to the __enter__ and __exit__ callables as an error:

```
contextmanaged.py:13:0: E1101: Generator 'generator' has no '__enter__' member (no-member)
contextmanaged.py:14:0: E1101: Generator 'generator' has no '__exit__' member (no-member)
```

while "uncommon" at least, I think it should not be marked as an error.

```
> pylint --version
pylint 2.1.1
astroid 2.0.4
Python 3.6.3 (v3.6.3:2c5fed8, Oct  3 2017, 18:11:49) [MSC v.1900 64 bit (AMD64)]
```

--- a/tests/functional/r/regression_02/regression_2567.py
+++ b/tests/functional/r/regression_02/regression_2567.py
@@ -1,0 +1,33 @@
+"""
+Regression test for `no-member`.
+See: https://github.com/PyCQA/pylint/issues/2567
+"""
+
+# pylint: disable=missing-docstring,too-few-public-methods
+
+import contextlib
+
+
+@contextlib.contextmanager
+def context_manager():
+    try:
+        yield
+    finally:
+        pass
+
+
+cm = context_manager()
+cm.__enter__()
+cm.__exit__(None, None, None)
+
+
+@contextlib.contextmanager
+def other_context_manager():
+    try:
+        yield
+    finally:
+        pass
+
+
+with other_context_manager():  # notice the function call
+    pass

